### ISSUE TITLE
C# Attributes

### ISSUE THREAD

**Issue by DeNelo** [PrismJS/prism#662]
(EDIT: And when I say attribute, I mean annotation... sorry)
In C#, decorating with annotations goes unnoticed by the parser. For instance, in:

```
[RemotableService]
[TaskDefinition("RightName", 
/*!localizationId*/"RightDescription.",/*!localizationId2*/"LocalRightName")]
public class RemotableExample : IRemotableExample
{
  ...
```

-the "[RemotableService]" annotation does not get colored at all. 
In "[TaskDefinition(...]", the word TaskDefinition gets colored, but not the [.
Is there anything to do about that..?

**Comment by Golmote** [comment]
@mvalipour Could you take a look?

**Comment by nauzilus** [comment]
I've also noticed it doesn't highlight generics:
```
public void Foo() { Bar(); }
public void Foo<T>() { Bar<T>(); }
```
The method name and invocation will highlight on the first line, but not the second.

I started to look into these a while ago but got busy :(

I'll try and have a look again if I can get time (and it's not tended to in the meantime).

**Comment by mvalipour** [comment]
well, attribute selection is going to be tricky because there can be confusion between `[Attribute]` and the c#6's dictionary initialization:

```
var dic = new Dictionary<string, string> {
    [key] = value,
    [key2] = value2
}
```

for generics tho, it should be fine. do you guys agree?

**Comment by Golmote** [comment]
@mvalipour: Can you submit a PR if you think you can improve the current highlighting?

**Comment by mvalipour** [comment]
I'm happy to do so, however, what I'm trying to figure out is that this is something within the possibilities of prism. -- to my understanding, prism is very limited in terms of syntaxes it can find (given it doesn't parse the code and purely relies on regular expressions).

I had a quick look and nothing seems to be done around for example templates in cpp (which are the equivalent of c# generics)

**Comment by Golmote** [comment]
Regexp are powerful though. The idea here is trying to achieve the best possible result using them. :)

**Comment by Golmote** [comment]
Fixes for #1365 and #1371 seem to have fixed this issue too, so i'm closing it.
